src/CompilationInfo.h

```diff
@@ -6,10 +6,32 @@
 // File `CompilationInfo.cpp` which is created and linked by CMake.
 
 #pragma once
+#include <atomic>
 #include <string_view>
+
+#include "util/Synchronized.h"
 namespace qlever::version {
-// Short version of the hash of the commit that was used to QLever.
+
+// The following two constants require linking against the `compilationInfo`
+// library which is recreated on every compilation. Short version of the hash of
+
+// The commit that was used to compile QLever.
 extern const std::string_view GitShortHash;
 // The date and time at which QLever was compiled.
 extern const std::string_view DatetimeOfCompilation;
+
+// The following two versions of the above constants do NOT require linking
+// against the `compilationInfo` library, but only the inclusion of this header.
+// They only have meaningful values once the `copyVersionInfo` function (below)
+// was called. This is currently done in the `main` functions of
+// `IndexBuilderMain.cpp` and `ServerMain.cpp`.
+inline ad_utility::Synchronized<std::string_view> gitShortHashWithoutLinking{
+    std::string_view{"git short hash not set"}};
+inline ad_utility::Synchronized<std::string_view>
+    datetimeOfCompilationWithoutLinking{
+        std::string_view{"git short hash not set"}};
```

Comment:
I'm gonna be honest with you: In general it's good practice to guard stuff like this against concurrent access. But I think in this particular case we can get away with just using a regular string_view and pinky promise that we will only ever write to this variable before any non-main threads are spawned.
Maybe put a big warning there just be be sure.

Reply:
I don't see any disadvantage here to doing it the right way
(if you need to have global mutable variables, at least make them threadsafe).
We only have very few usages of this, and the performance overhead also is a non-issue, so having to write `.wlock()->` three times is not really a problem.